src/fu-common.c

```diff
@@ -349,6 +349,47 @@ fu_common_add_argv (GPtrArray *argv, const gchar *fmt, ...)
 		g_ptr_array_add (argv, g_strdup (split[i]));
 }
 
+static gchar *
+fu_common_find_program_in_path (const gchar *basename, GError **error)
+{
+	gchar *fn = g_find_program_in_path (basename);
+	if (fn == NULL) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "missing executable %s in PATH",
+			     basename);
+		return NULL;
+	}
+	return fn;
+}
+
+static gboolean
+fu_common_test_namespace_support (GError **error)
+{
+	/* test if CONFIG_USER_NS is valid */
+	if (!g_file_test ("/proc/self/ns/user", G_FILE_TEST_IS_SYMLINK)) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "missing CONFIG_USER_NS in kernel");
+		return FALSE;
+	}
+	if (g_file_test ("/proc/sys/kernel/unprivileged_userns_clone", G_FILE_TEST_EXISTS)) {
+		g_autofree gchar *clone = NULL;
+		if (!g_file_get_contents ("/proc/sys/kernel/unprivileged_userns_clone", &clone, NULL, error))
+			return FALSE;
+		if (g_ascii_strtoll (clone, NULL, 10) == 0) {
+			g_set_error (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_NOT_SUPPORTED,
+				     "unprivileged user namespace clones disabled by distro");
+			return FALSE;
+		}
+	}
+	return TRUE;
+}
+
```

Comment:
You split these out to use them for this PR, but didn't use to check if curl actually in path.

Reply:
Hah, indeed. Will fix.